### ISSUE TITLE
Support bingo

### ISSUE THREAD

**Issue by joshhunt** [evilmartians/lefthook#509]
### :zap: Summary

[We're trying to migrate to lefthook](https://github.com/grafana/grafana/pull/66608), using [bingo](https://github.com/bwplotka/bingo) to manage the lefthook binary dependency. The neat thing about using bingo is we can instruct contributors "run `make lefthook-install`" and the required version of lefthook will be downloaded and the hooks installed in the one step^1

The obstacle with using lefthook and bingo together is that bingo installs lefthook onto $PATH with a versioned name (like `lefthook-v1.4.3`), which the [lefthook git hooks](https://github.com/evilmartians/lefthook/blob/master/internal/templates/hook.tmpl) cannot find. I have tried to use lefthook's `rc` option to provide a lefthook function that points to the correct place, but the hook says it cannot find the file - it appears this should be an absolute path to a file, rather than relative to the project root.

I can think of two ways to support bingo:

 - **Support bingo explicitly**. Test if [`.bingo/variables.env`](https://github.com/grafana/grafana/blob/main/.bingo/variables.env) exists, `source` it, and then run `${LEFTHOOK}`.
 - **Allow relative path to rc file**. If we can commit a `lefthook.rc` file in the repo and reference that in the lefthook config, we can do the required bootstraping to make a `lefthook` binary available to the git hook. 

If either of these are acceptable I'm willing to contribute the change myself (though neither bash nor golang are my strong suits so I might need some pointers)

---

1: Why not use the npm package? Our goal of moving to lefthook is to remove the yarn dependency on committing, and make the hooks opt-in rather than forced onto all contributors

**Comment by joshhunt** [comment]
This commit https://github.com/grafana/grafana/pull/66608/commits/af549c72 is more or less what I expected to do to get bingo and lefthook to work with each other. Note the `.bingo/variables.env` file which contains the path to lefthook.

**Comment by mrexox** [comment]
Hey @joshhunt! Thank you for the feature request. I will consider extending the hook template with the check for `.bingo/variables.env`, I just need to dig into this tool. If you have time and will you can provide a PR that extends [hook template](https://github.com/evilmartians/lefthook/blob/master/internal/templates/hook.tmpl) file (I think it's pretty straightforward what's going on in there).

Anyway, releasing this feature will take some time, and I can suggest you a workaround for the relative path to rc file:

```yml
# lefthook.yml

rc: ./lefthook.rc # note, ./ - is important here since it won't work without it
```

```bash
# lefthook.rc

. .bingo/variables.env

function lefthook() {
  ${LEFTHOOK} $@
}
```

This way lefthook will be available for git hooks as a shell function. Not a perfect option but working. Please, write back about any success on this :)

**Comment by joshhunt** [comment]
Thanks for the update @mrexox. I had actually tried setting `rc: ./lefthook.rc` (with the leading `./`) before and it didn't work, but I don't think I reinstalled the git hooks in between attempts so i never actually got the benefit of that.

With the lefthook.rc file now working for us, I'm happy to close this issue. While built-in bingo support might be neat, I think rc file is a more than acceptable way to support tools like that.
